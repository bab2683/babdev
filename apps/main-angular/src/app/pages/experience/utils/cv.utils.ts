import { CV } from '@models';
import { calculateDuration } from '@utils';

export function parseCVDates(cv: CV): CV {
  cv.sections.experiences.items.forEach((exp) => {
    exp.dates.start = new Date(exp.dates.start);
    if (exp.dates.end) {
      exp.current = false;
      exp.dates.end = new Date(exp.dates.end);
    } else {
      exp.current = true;
      exp.dates.end = new Date();
    }
    exp.duration = calculateDuration(exp.dates.start, exp.dates.end);
  });
  return cv;
}
