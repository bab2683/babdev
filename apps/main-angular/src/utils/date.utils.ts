import { Duration } from '@models';

export const averageDaysPerMonth: number = 30;
export const daysFromMS: number = 1000 * 60 * 60 * 24;

export function calculateDuration(start: Date, end: Date): Duration {
  const days: number = Math.floor(
    (end.getTime() - start.getTime()) / daysFromMS
  );

  if (days > 365) {
    const years: number = Math.floor(days / 365);

    return {
      months: Math.floor((days - years * 365) / averageDaysPerMonth),
      years
    };
  } else {
    return {
      months: Math.floor(days / averageDaysPerMonth),
      years: null
    };
  }
}
