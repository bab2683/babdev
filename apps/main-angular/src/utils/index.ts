export * from './date.utils';
export * from './page.utils';
export * from './string.utils';
