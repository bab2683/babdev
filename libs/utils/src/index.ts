export * from './lib/css-variables.utils';
