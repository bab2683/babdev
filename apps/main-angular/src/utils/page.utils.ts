import { Page } from '@models';
import { capitalize } from './string.utils';

export const generatePageModel = ({
  name,
  hasBg,
  isHome
}: {
  name: string;
  hasBg: boolean;
  isHome: boolean;
}): Page => ({
  data: {
    ...(hasBg ? { bgPath: `bg/${name}` } : {}),
    dictionary: {
      location: `/pages/${name}/`,
      name
    },
    name,
    titleKey: `common.PAGES.${name.toUpperCase()}`
  },
  moduleName: `${capitalize(name)}Module`,
  path: isHome ? '' : name
});
