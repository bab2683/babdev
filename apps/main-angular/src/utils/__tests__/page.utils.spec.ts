import { generatePageModel } from '../page.utils';

describe('Page utils', () => {
  describe('generatePageModel', () => {
    it('should match snapshot for homepage', () => {
      const result = generatePageModel({
        name: 'home',
        hasBg: false,
        isHome: true
      });

      expect(result).toMatchSnapshot();
    });

    it('should match snapshot for landing page', () => {
      const result = generatePageModel({
        name: 'experience',
        hasBg: true,
        isHome: false
      });

      expect(result).toMatchSnapshot();
    });
  });
});
