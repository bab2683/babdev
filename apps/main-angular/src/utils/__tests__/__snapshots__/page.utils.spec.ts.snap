// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page utils generatePageModel should match snapshot for homepage 1`] = `
Object {
  "data": Object {
    "dictionary": Object {
      "location": "/pages/home/",
      "name": "home",
    },
    "name": "home",
    "titleKey": "common.PAGES.HOME",
  },
  "moduleName": "HomeModule",
  "path": "",
}
`;

exports[`Page utils generatePageModel should match snapshot for landing page 1`] = `
Object {
  "data": Object {
    "bgPath": "bg/experience",
    "dictionary": Object {
      "location": "/pages/experience/",
      "name": "experience",
    },
    "name": "experience",
    "titleKey": "common.PAGES.EXPERIENCE",
  },
  "moduleName": "ExperienceModule",
  "path": "experience",
}
`;
